edx_event_bus_kafka/internal/producer.py

```diff
@@ -167,18 +173,22 @@ def inner_to_dict(event_data, ctx=None):  # pylint: disable=unused-argument
     return key_serializer, value_serializer
 
 
+@attr.s(kw_only=True, repr=False)
 class ProducingContext:
     """
     Wrapper class to allow us to link a call to produce() with the on_event_deliver callback
     """
-    def __init__(self, context_data):
-        self.context_data = context_data
-
-    def __str__(self):
-        return " ".join([f"{key}={value}" for key, value in self.context_data.items()])
-
-    def set_context_data(self, key, value):
-        self.context_data[key] = value
+    full_topic = attr.ib(type=str, default=None)
+    event_type = attr.ib(type=str, default=None)
+    event_key = attr.ib(type=str, default=None)
+    signal = attr.ib(type=OpenEdxPublicSignal, default=None)
+    initial_topic = attr.ib(type=str, default=None)
+    event_key_field = attr.ib(type=str, default=None)
+    event_data = attr.ib(type=dict, default=None)
+
+    def __repr__(self):
+        """Create a splunk-friendly string"""
```

Comment:
Splunk is 2U-specific -- perhaps this?
```suggestion
        """Create a logging-friendly string"""
```